### PR TITLE
Tasks 27.06 

### PR DESCRIPTION
**Pull request by pavlentiysv**

### CODE REVIEW

File: js/27.06/task10.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
+function isEqual(obj1, obj2) {
```
Comment (angelinabylitskaya) [inline js/27.06/task10.js]: Won't work if objects will be the same but with different fields order.

File: js/27.06/task2.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+function difference(arr1, arr2) {
```
Comment (angelinabylitskaya) [inline js/27.06/task2.js]: The code can be simplified.
Amount if iterations should be lessened.

File: js/27.06/task4.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+function flattenDeep(arr) {
```
Comment (angelinabylitskaya) [inline js/27.06/task4.js]: The code can be simplified.
Try to use "flat" method or other optimization.

File: js/27.06/task6.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+function uniq(arr) {
```
Comment (angelinabylitskaya) [inline js/27.06/task6.js]: Try to use reduce + filter here.

File: js/27.06/task9.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+function groupBy(arr, key) {
```
Comment (angelinabylitskaya) [inline js/27.06/task9.js]: Refactor the code to be more readable.

File: js/27.06/task2.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+  var uniqArr1 = [...new Set(arr1)];
```
Comment (kobweb) [inline js/27.06/task2.js]: Let's do it without `Set`

File: js/27.06/task4.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+  var arrayTrigger = 1;
```
Comment (kobweb) [inline js/27.06/task4.js]: Try to avoid using of `var`. It's better to use `let` instead.

File: js/27.06/task4.js [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+  while (arrayTrigger > 0) {
+    arrayTrigger = 0;
+    for (var i = 0; i < arr.length; i++) {
```
Comment (kobweb) [inline js/27.06/task4.js]: You should avoid usage of nested loops. Try to optimize your code.

File: js/27.06/task2.js (hunk 6 above)
Comment (pavlentiysv) [reply js/27.06/task2.js]: It was done without `Set` in my first commit. Angelina adviced to use `Set` and I changed it.

File: js/27.06/task6.js [hunk 9]
```diff
@@ -0,0 +1,10 @@
...
+  let resArr = arr.filter(function(currentValue, currentIndex) {
+    return arr.indexOf(currentValue) === arr.lastIndexOf(currentValue);
+  });
+
+  console.log(resArr);
+  return resArr;
+}
+
```
Comment (ElizavetaSelivanova) [inline js/27.06/task6.js]: Read about `const` and `let`

File: js/27.06/task10.js (hunk 1 above)
Comment (angelinabylitskaya) [reply js/27.06/task10.js]: Please, update function.

File: js/27.06/task3.js [hunk 10]
```diff
@@ -0,0 +1,15 @@
+function findIndex(objectForSearch, elementToFind) {
+  const isFunction = typeof elementToFind === 'function';
+  const searchMethod = isFunction ? 'findIndex' : 'indexOf';
+  return (objectForSearch[searchMethod](elementToFind));
```
Comment (angelinabylitskaya) [inline js/27.06/task3.js]: Redundant braces.

File: js/27.06/task9.js [hunk 11]
```diff
@@ -0,0 +1,10 @@
...
+  return arr.reduce(function(accumulator, currentValue){
```
Comment (angelinabylitskaya) [inline js/27.06/task9.js]:
```
[].ruduce((acc, currentValue) => {
	const key = getKey(currentValue);
	return {
		...acc,
		[key]: [...(acc[key] || []), currentValue]
	};
})
```

File: js/27.06/task2.js [hunk 12]
```diff
@@ -0,0 +1,15 @@
+function difference(arr1, arr2) {
+  const commonArr = arr1.concat(arr2);
+  let bufObject = commonArr.reduce(function(accumulator, currentValue) {
+    accumulator.hasOwnProperty(currentValue) ? accumulator[currentValue]++ : (accumulator[currentValue] = 1);
```
Comment (angelinabylitskaya) [inline js/27.06/task2.js]: It is better to use if else here.

File: js/27.06/task5.js [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+  return arr.reduce(function(accumulator, currentValue) {
+    const [key, value] = currentValue;
+    return {
+      ...accumulator,
+      [key]: value
+    };
+  }, {});
+}
+
+//Previous solution
```
Comment (angelinabylitskaya) [inline js/27.06/task5.js]: Please, remove the previous solution.

**Pull request merged**